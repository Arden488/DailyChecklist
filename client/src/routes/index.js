import Questions from '@/components/pages/QuestionsPage.vue';
import Form from '@/components/pages/FormPage.vue';
import Overview from '@/components/pages/OverviewPage.vue';

const routes = [
  {
    path: '/',
    component: Overview
  },
  {
    path: '/questions',
    component: Questions
  },
  {
    path: '/form',
    component: Form
  }
];

export default routes;
