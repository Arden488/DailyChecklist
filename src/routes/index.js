import Questions from '@/components/pages/QuestionsPage.vue';
import CreateReport from '@/components/pages/CreateReportPage.vue';
import Overview from '@/components/pages/OverviewPage.vue';
import CreateQuestion from '@/components/pages/CreateQuestionPage.vue';
import EditQuestion from '@/components/pages/EditQuestionPage.vue';
import SignInPage from '@/components/pages/SignInPage.vue'

const routes = [
  {
    path: '/signin',
    name: 'Sign In',
    component: SignInPage,
    meta: {
      layout: 'clean'
    }
  },
  {
    path: '/',
    name: 'Overview',
    component: Overview,
    meta: {
      requiresAuth: true
    }
  },
  {
    path: '/questions',
    name: 'Questions',
    component: Questions,
    meta: {
      requiresAuth: true
    }
  },
  {
    path: '/questions/create',
    name: 'CreateQuestion',
    component: CreateQuestion,
    meta: {
      requiresAuth: true
    }
  },
  {
    path: '/questions/:id',
    name: 'EditQuestion',
    component: EditQuestion,
    meta: {
      requiresAuth: true
    }
  },
  {
    path: '/reports/create',
    name: 'CreateReport',
    component: CreateReport,
    meta: {
      requiresAuth: true
    }
  }
];

export default routes;
