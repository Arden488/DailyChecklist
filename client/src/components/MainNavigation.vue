<template>
  <el-menu
    :router="true" 
    :default-active="$route.path"
    class="header-navigation">
    
    <el-menu-item 
      class="header-navigation__item" 
      index="/" 
      :route="'/'"
    >
      Overview
    </el-menu-item>
    <el-menu-item 
      class="header-navigation__item" 
      index="/reports/create" 
      :route="'/reports/create'"
    >
      Submit report
    </el-menu-item>
    <el-submenu index="">
      <template slot="title">Questions</template>
      <el-menu-item 
        class="header-navigation__item" 
        index="/questions" 
        :route="'/questions'"
      >
        List
      </el-menu-item>
      <el-menu-item 
        class="header-navigation__item" 
        index="/questions/create" 
        :route="'/questions/create'"
      >
        Create
      </el-menu-item>
    </el-submenu>

  </el-menu>
</template>

<script>
import routes from '@/routes';

export default {
  name: 'MainNavigation',
  data () {
    return {
      routes
    }
  }
}
</script>

<style>
.header-navigation__item {
  text-decoration: none;
}
</style>
