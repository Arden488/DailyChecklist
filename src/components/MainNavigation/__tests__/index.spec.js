/* global it, describe, expect */

import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import Vue from 'vue';
import VueRouter from 'vue-router';
import ElementUI from 'element-ui';
import MainNavigation from '../MainNavigation.vue';
import routes from '@/routes';

const localVue = createLocalVue();
localVue.use(ElementUI);
localVue.use(VueRouter);
const router = new VueRouter({ routes });

describe('MainNavigation component', () => {
  it('is a Vue instance', () => {
    const wrapper = shallowMount(MainNavigation, { localVue, router });
    expect(wrapper.isVueInstance).toBeTruthy()
  })

  it('is renders correctly', () => {
    const wrapper = shallowMount(MainNavigation, { localVue, router });
    expect(wrapper).toMatchSnapshot()
  })
});
