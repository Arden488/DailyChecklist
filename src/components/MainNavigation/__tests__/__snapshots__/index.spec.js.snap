// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavigation component is renders correctly 1`] = `
<ul role="menubar" class="header-navigation el-menu">
  <li role="menuitem" tabindex="-1" class="el-menu-item header-navigation__item is-active" style="padding-left: 20px;">
    Overview
  </li>
  <li role="menuitem" tabindex="-1" class="el-menu-item header-navigation__item" style="padding-left: 20px;">
    Submit report
  </li>
  <li role="menuitem" aria-haspopup="true" class="el-submenu">
    <div class="el-submenu__title" style="padding-left: 20px;">Questions
      <i class="el-submenu__icon-arrow el-icon-arrow-down"></i>
    </div>
    <ul role="menu" class="el-menu el-menu--inline" style="display: none;">
      <li role="menuitem" tabindex="-1" class="el-menu-item header-navigation__item" style="padding-left: 40px;">
        List
      </li>
      <li role="menuitem" tabindex="-1" class="el-menu-item header-navigation__item" style="padding-left: 40px;">
        Create
      </li>
    </ul>
  </li>
</ul>
`;
